source/plugins/ruby/in_containerinventory.rb

```diff
@@ -63,10 +64,14 @@ def enumerate
       $log.info("in_container_inventory::enumerate : Begin processing @ #{Time.now.utc.iso8601}")
       if ExtensionUtils.isAADMSIAuthMode()
         $log.info("in_container_inventory::enumerate: AAD AUTH MSI MODE")
-        if @tag.nil? || !@tag.start_with?(Constants::EXTENSION_OUTPUT_STREAM_ID_TAG_PREFIX)
-          @tag = ExtensionUtils.getOutputStreamId(Constants::CONTAINER_INVENTORY_DATA_TYPE)
+        @tag, isFromCache = KubernetesApiClient.getOutputStreamIdAndSource(Constants::CONTAINER_INVENTORY_DATA_TYPE, @tag, @agentConfigRefreshTracker)
```

Comment:
All of this is also common to all the ruby files which could have been made modular. But, since we will be deprecating ruby and fluentd soon, we can let this be

Reply:
> All of this is also common to all the ruby files which could have been made modular. But, since we will be deprecating ruby and fluentd soon, we can let this be

We would need  getting stream id  logic in all the plugins. I dont see any better way to share the stream id information across alll the plugins without implementing reading of extension config in each plugin which will not performant.